hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/qjournal/client/TestQuorumJournalManager.java

```diff
@@ -1171,6 +1176,21 @@ public void testSelectViaRpcAfterJNRestart() throws Exception {
     }
   }
 
+  /**
+   * Tests to throw an exception if the jn maintenance nodes exceeds half of the journalnode number.
+   */
+  @Test
+  public void testJNMaintenanceListViaRpcTwoJNsError() throws Exception {
+    StringJoiner maintenanceListBuff = new StringJoiner(",");
+    for (int i = 0; i < 2; i++) {
+      maintenanceListBuff.add(
+          NetUtils.getHostPortString(cluster.getJournalNode(i).getBoundIpcAddress()));
+    }
+    this.conf.set(DFS_JOURNALNODE_MAINTENANCE_NODES_KEY, maintenanceListBuff.toString());
+    assertThrows(IllegalArgumentException.class, () -> createSpyingQJM());
+  }
+
+
```

Comment:
I assume we also need a test case to to verify maintenance journal nodes are indeed excluded? total JN=3, with 1 is excluded. after we initialize, verify only 2 JNs are included.